### PR TITLE
Add a warning dialog when opening the old sample app

### PR DESCRIPTION
**Pull request by zsmb13**
### 🎯 Goal

Making sure that nobody accidentally runs the old sample instead of the new one.

### 🛠 Implementation details

Self-explanatory code.

### 🧪 Testing

Open old sample.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] ~Changelog is updated with client-facing changes~
- [ ] ~New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- [ ] ~Affected documentation updated (CMS, cookbooks, tutorial)~
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt [hunk 1]
```diff
@@ -19,4 +30,44 @@ class StartupFragment : Fragment() {
...
+
+        if (warningShown(context)) {
+            return
+        }
+
+        val checkbox = CheckBox(context).apply { text = "Do not show this again" }
+        val frame = FrameLayout(context).apply {
+            setPadding(Utils.dpToPx(16), Utils.dpToPx(12), Utils.dpToPx(16), 0)
+        }
+        frame.addView(checkbox)
```
Comment (zsmb13) [inline stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt:45]: Artisinal, organic, hand-crafted custom layout

Comment (petyosi) [reply stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt:45]: You forgot gluten free

Comment (zsmb13) [reply stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt:45]: That also happens to be true!

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt [hunk 2]
```diff
@@ -19,4 +30,44 @@ class StartupFragment : Fragment() {
...
+
+    private fun warningShown(context: Context): Boolean {
+        return PreferenceManager.getDefaultSharedPreferences(context)
+            .getBoolean(OLD_SAMPLE_WARNING_SHOWN_KEY, false)
+    }
+
+    private fun setWarningShown(context: Context) {
+        PreferenceManager.getDefaultSharedPreferences(context).edit {
+            putBoolean(OLD_SAMPLE_WARNING_SHOWN_KEY, true)
+        }
```
Comment (bychkovdmitry) [inline stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt:67]: Does this work without `apply()`?

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt:45]: Looks like a code from somebody who started to learn Compose 😅

**Review by bychkovdmitry — APPROVED** [review]

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt (hunk 2 above)
Comment (zsmb13) [reply stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/startup/StartupFragment.kt:67]: Yes! https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:core/core-ktx/src/main/java/androidx/core/content/SharedPreferences.kt;l=39

**Pull request merged**